### DISCUSSION TITLE
using PBC for npz dataset

### DISCUSSION THREAD

**Discussion by ken1944** [mir-group/nequip#242] [Q&A] [answered]
Hello,

I tried finding this in the documentation and discussions but wanted to confirm. I'm using NequIP and training a model with a dataset that uses "pbc". My data format is npz and I'm specifying a pbc array in the npz file like `pbc = np.array([19.5894, 19.5542, 19.5796], dtype='f')` with the x,y and z dimensions. I wanted to confirm if this was the right way to mention the pbc since the output file (with and without pbc) shows pbc in the form pbc=[nframe,3]

Thanks for the help!

**Comment by Linux-cpp-lisp** [comment, marked as answer]
Hi @ken1944 ,

Thanks for your interest in our code!

We distinguish between `pbc` (_whether_ each of the x,y,z directions is periodic) and `cell`, which gives the box that is periodic as a `(3, 3)` shape tensor whose rows are the three xyz cell vectors. So you'd want to set `pbc = (True, True, True)` and `cell = [[19.57, 0, 0], [0, 19,57, 0], [0, 0, 19.57]]`, basically.

**Reply by ken1944** [reply]
Hi @Linux-cpp-lisp,

Thanks for your response! The model seems to train with pbc now. I also went ahead and added `pbc` and `cell` under `npz_fixed_field_keys` as mentioned in `full.yaml`

Thanks again!

**Reply by Linux-cpp-lisp** [reply]
👍 

Yes, exactly: when all examples in our dataset have the same `pbc`/`cell`, that's what `fixed_fields` is for.
